include/forcing/NetCDFPerFeatureDataProvider.hpp

```diff
@@ -0,0 +1,464 @@
+            auto num_times = nc_file->getDim("time").getSize();
+
+            // allocate storage for the raw time array
+            std::vector<double> raw_time;
+            raw_time.resize(num_times);
+
+            // get the time variable
+            auto time_var = nc_file->getVar("Time");
+
+            // read from the first catchment row to get the recorded times
+            std::vector<size_t> start;
+            start.push_back(0);
+            start.push_back(0);
+            std::vector<size_t> count;
+            count.push_back(1);
+            count.push_back(num_times);
+            time_var.getVar(start, count, &raw_time[0]);
+
+            // read the meta data to get the time_unit
+            auto time_unit_att = time_var.getAtt("units");
+
+            // if time att is not encoded 
+            // TODO determine how this should be handled
+            std::string time_unit_str;
+
+            if ( time_unit_att.isNull() )
+            {
+                time_unit = TIME_SECONDS;
+            }
+            else
+            {  
+                time_unit_att.getValues(time_unit_str);
+            }
+
+            if ( time_unit_str == "h" || time_unit_str == "hours ")
+                time_unit = TIME_HOURS;
+            else if ( time_unit_str == "m" || time_unit_str == "minuets" )
+                    time_unit = TIME_MINUETS;
+            else if ( time_unit_str ==  "s" || time_unit_str == "seconds" )
+                    time_unit = TIME_SECONDS;
+            else if ( time_unit_str ==  "ms" || time_unit_str == "miliseconds" )
+                    time_unit = TIME_SECONDS;
+            else if ( time_unit_str ==  "us" || time_unit_str == "microseconds" )
+                    time_unit = TIME_SECONDS;
+            else if ( time_unit_str ==  "ns" || time_unit_str == "nanoseconds" )
+                    time_unit = TIME_SECONDS;
+            
+            double time_scale_factor;
+
+            switch (time_unit)
+            {
+                case TIME_HOURS:
+                    time_scale_factor = 3600;
+                break;
+
+                case TIME_MINUETS:
+                    time_scale_factor = 60;
+                break;
+
+                case TIME_SECONDS:
+                    time_scale_factor = 1;
+                break;
+
+                case TIME_MILLISECONDS:
+                    time_scale_factor = .001;
+                break;
+
+                case TIME_MICROSECONDS:
+                    time_scale_factor = .000001;
+                break;
+
+                case TIME_NANOSECONDS:
+                    time_scale_factor = .000000001;
+                break;
+            }
+
+            // read the meta data to get the epoc start
+            auto epoc_att = time_var.getAtt("epoc_start");
+            std::string epoc_start_str;
+
+            if ( epoc_att.isNull() )
+            {
+                epoc_start_str = "01/01/1970 00:00:00";
```

Comment:
This should probably log a warning and indicate the "default epoc start time"

Reply:
I agree, I suppose the data provider should be constructed with a logging interface?